src/pages-and-resources/discussions/app-config-form/utils.js

```diff
@@ -1,5 +1,66 @@
-const filterItemFromObject = (array, key, value) => (
+import moment from 'moment';
+import _ from 'lodash';
+
+import { blackoutDatesStatus as constants } from '../data/constants';
+
+export const filterItemFromObject = (array, key, value) => (
   array.filter(item => item[key] !== value)
 );
 
-export default filterItemFromObject;
+export const checkFieldErrors = (touched, errors, field, propertyName, index) => Boolean(
+  touched[field] && touched[field][index] && touched[field][index][propertyName]
+  && errors[field] && errors[field][index] && errors[field][index][propertyName],
+);
+
+export const checkStatus = ([startDate, endDate]) => {
+  const today = moment(); let status;
+
+  if (moment(endDate).isBefore(today, 'days')) {
+    status = constants.COMPLETE;
+  } else if (moment(startDate).isAfter(today, 'days')) {
+    status = constants.UPCOMING;
+  } else {
+    status = constants.ACTIVE;
+  }
+  return status;
+};
+
+export const formatDate = (date, time) => (time ? `${date}T${time}` : date);
+export const isSameDay = (startDate, endDate) => moment(startDate).isSame(endDate, 'day');
+export const isSameMonth = (startDate, endDate) => moment(startDate).isSame(endDate, 'month');
+export const isSameYear = (startDate, endDate) => moment(startDate).isSame(endDate, 'year');
+
+export const sortBlackoutDatesByStatus = (data, status, order) => (
+  _.orderBy(data.filter(date => date.status === status),
+    [(obj) => moment(formatDate(obj.startDate, obj.startTime))], [order])
+);
+
+export const formatBlackoutDates = ({
+  startDate, startTime, endDate, endTime,
+}) => {
+  let formattedDate;
+  const hasSameDay = isSameDay(startDate, endDate);
+  const hasSameMonth = isSameMonth(startDate, endDate);
+  const hasSameYear = isSameYear(startDate, endDate);
+  const isTimeAvailable = Boolean(startTime && endTime);
```

Comment:
to cleanup you can introduce these variables to avoid reputation. 
```
const mStartDate = moment(startDate);
const mEndDate = moment(endDate);
const mStartDateTime = moment(`${startDate}T${startTime}`);
const mEndDateTime = moment(`${endDate}T${startTime}`);
```